AdvancedChangedFiles/Experience.jsx

```diff
@@ -0,0 +1,223 @@
+                                                            cursor: 'pointer'
+                                                        }}
+                                                        onClick={this.handleEditClick.bind(this, experience)}
+                                                    >
+                                                        ✏️
+                                                    </button>
+                                                    <button
+                                                        style={{
+                                                            color: 'black',
+                                                            background: 'none',
+                                                            border: 'none',
+                                                            cursor: 'pointer'
+                                                        }}
+                                                        onClick={this.handleDeleteClick.bind(this, experience.id)}
+                                                    >
+                                                        ❌
+                                                    </button>
+                                                </td>
+                                            </React.Fragment>
+                                        )}
+                                    </tr>
+                                ))
+                            )}
+                            {isAdding && (
+                                <tr>
+                                    <td><input type="text" name="company" placeholder="Add Company" value={newExperience.company} onChange={this.handleChange} style={{ width: '100%' }} /></td>
+                                    <td><input type="text" name="position" placeholder="Add Position" value={newExperience.position} onChange={this.handleChange} style={{ width: '100%' }} /></td>
+                                    <td><input type="text" name="responsibilities" placeholder="Add Responsibilities" value={newExperience.responsibilities} onChange={this.handleChange} style={{ width: '100%' }} /></td>
+                                    <td><input type="date" name="startDate" value={newExperience.startDate} onChange={this.handleChange} style={{ width: '100%' }} /></td>
+                                    <td><input type="date" name="endDate" value={newExperience.endDate} onChange={this.handleChange} style={{ width: '100%' }} /></td>
+                                    <td style={{ textAlign: 'center' }}>
+                                        <button className="ui teal button" onClick={this.handleSaveClick}>Add</button>
+                                        <button className="ui button" onClick={this.handleCancelClick}>Cancel</button>
+                                    </td>
+                                </tr>
+                            )}
```

Comment:
This might work but in terms of code optimisation, readability, and managing, we can enhance this. Instead of implementing multiple ternary conditions here, can you create a function for them and call/use them accordingly?